### PR TITLE
[Matrix] fix for large media sequences

### PR DESCRIPTION
**Pull request by glennguy**
Backport of #878

### CODE REVIEW

File: src/parser/DASHTree.cpp [hunk 1]
```diff
@@ -91,7 +91,7 @@ static unsigned int ParseSegmentTemplate(const char** attr,
                                          std::string baseURL,
```
Comment (CastagnaIT) [inline src/parser/DASHTree.cpp:91]: miss to change ParseSegmentTemplate method return type to uint64_t

File: src/common/AdaptiveTree.h [hunk 2]
```diff
@@ -245,9 +245,11 @@ class ATTRIBUTE_HIDDEN AdaptiveTree
       return get_segment_pos(current_segment_);
     };
 
-    uint32_t getCurrentSegmentNumber() const
+    uint64_t getCurrentSegmentNumber() const
     {
-      return current_segment_ ? get_segment_pos(current_segment_) + startNumber_ : ~0U;
+      return current_segment_
+                 ? static_cast<uint64_t>(get_segment_pos(current_segment_) + startNumber_)
+                 : ~0U;
```
Comment (CastagnaIT) [inline src/common/AdaptiveTree.h]: `~0U;` to `~0ULL;`

**Review by CastagnaIT — COMMENTED** [review]
two minor things

**Comment by glennguy** [comment]
Thanks @CastagnaIT!

**Review by CastagnaIT — APPROVED** [review]

**Comment by gizmocuz** [comment]
ParseSegmentTemplate returns uint64_t, but `AdaptationSet::startNumber_` is still a unsigned int, is this intended ? In this case maybe cast the return to `(unsigned int)` in DashTree.cpp

**Comment by glennguy** [comment]
@gizmocuz my bad, I overlooked that variable in the backport.  Did you want to make a PR or shall I?

**Comment by gizmocuz** [comment]
@glennguy , I leave this all to you as you did already most of the work ;-)

**Pull request merged**